### ISSUE TITLE
Resized images have distorted background colors (jpg)

### ISSUE THREAD

**Issue by ioweb-gr** [magento/magento2#22080]
### Preconditions (*)
1. Magento 2.3.1 using Luma. It also happens on 2.2.7, 2.2.8, 2.3.0 
2. Tested using GD to resize.
3. Debian 9
4. PHP 7.2.16

### Steps to reproduce (*)

Test 1
1. Create a new installation of Magento 2.3.1 using Luma theme. 
2. Create a simple product. 
3. Add the attached images which have completely white background and various dimensions. 
![pyjamas2](https://user-images.githubusercontent.com/20220341/55309625-127c2000-5467-11e9-9f97-09b48bf19087.jpg)
![pyjamas1000](https://user-images.githubusercontent.com/20220341/55309626-127c2000-5467-11e9-8205-efab61f3e499.jpg)
![pyjamas1500](https://user-images.githubusercontent.com/20220341/55309628-1314b680-5467-11e9-9406-2ad25bb3dd84.jpg)
![pyjamas2000](https://user-images.githubusercontent.com/20220341/55309629-1314b680-5467-11e9-9332-b7d7b3845a3c.jpg)
4. Choose any quality and any type of image upload configuration in Stores -> Configuration -> System -> Image Upload Configuration
For testing purposes
![image](https://user-images.githubusercontent.com/20220341/55309755-62f37d80-5467-11e9-934c-6d749c4af6d9.png)
5. Go to  frontend and check the product. 
6. The image background has changed in some areas from #FFF to #FEFEFE and #FDFDFD 

Test 2
1. Follow steps 1-2. 
2. Create a completely white image in photoshop.
3. follow steps 4-6  
4. You will notice stripes appearing in the image.  

### Expected result (*)
1. The image should be keeping the white background as is.

### Actual result (*)
1. The image has areas where the background color is not white although in the original image it was. 
2. In properly calibrated monitors or monitors with bad quality the color distortion is really visible. Especially in non direct angles depending on monitors it appears extremely weird.

Check the animated gif for the resized image color issue and the original image compared. 
![2019-04-01_10-23-57](https://user-images.githubusercontent.com/20220341/55310264-b0241f00-5468-11e9-8760-446713b851ee.gif)

**Comment by ioweb-gr** [comment]
@magento-engcom-team give me 2.3-develop instance - upcoming 2.3.x release

**Comment by magento-engcom-team** [comment]
Hi @ioweb-gr. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @ioweb-gr, here is your [Magento instance](https://i-22080-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-22080-2-3-develop.instances.magento-community.engineering/admin](https://i-22080-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by ioweb-gr** [comment]
Yes, I was able to reproduce it on the provided instance. The problem was less visible but it's still there. 

![image](https://user-images.githubusercontent.com/20220341/55311051-b3200f00-546a-11e9-85f3-b2a36c5120a8.png)

PS: I wasn't able to click on the checkbox

**Comment by ananth-iyer** [comment]
@ioweb-gr I have found that default **Image Adapter: PHP GD2** is not working properly in your case.

Let us try with **Image Adapter: ImageMagick** on same Magento instance: https://i-22080-2-3-develop.instances.magento-community.engineering/admin

**Comment by ananth-iyer** [comment]
I have changed  **Image Adapter to ImageMagick**. Please try in Private Window and let us know

**Comment by ioweb-gr** [comment]
I just tried it and I can't reproduce on ImageMagick.

**Comment by ananth-iyer** [comment]
It is just the **PHP GD2** that is not able to process the images correctly.

This is why Magento has given 2 options for the image process - PHP GD2 (default), **ImageMagick** for better image process than PHP GD2

Your next steps are to:
1) Install **ImageMagick** on your server if not installed
2) Change **Image Adapter** to **ImageMagick**
3) On Cache Management page:
Flush Configuration & Full page cache
Click **Flush Catalog Images Cache** to clear older images and on product page visit it will generate the new image files.

So there is no issue with the Magento core functionality we can close the issue.

**Comment by ioweb-gr** [comment]
I understand what you mean that Magento offers two solutions but that doesn't mean that because one solution works, we should ignore the problem in the other one. 

Moreover in some hosting environments we have limited options where we cannot install ImageMagick. GD2 should still be working correctly. It's also the predefined method of image resize. 

Also I dont' think it's a problem with the GD extension itself  since Magento 1, Wordpress, Joomla, Aimeos and pretty much every other platform out there uses it without causing color distortion, so I'm inclined to say there's something wrong in Magento 2's implementation. Since we were able to reproduce it on a clean environment it should be investigated in my opinion.
